templates/tips_and_tricks.md

```diff
@@ -0,0 +1,49 @@
+# Lean Tips and Tricks
+
+This document contains some tips for intermediate and advanced users of Lean.
+It is currently focussed around Lean 3.
+
+### Commenting out large proofs using `sorry`
+
+Sometimes commenting out large chunks of a proof is helpful, while the default comment characters are `--` or `/- multi-line comment-/`, however often a more convenient way is to temporarily skip a long subproof (of a focussed goal surrounded by braces) by prepending `sorry;`
+```
+sorry; { rw blah,
+  ...
+  HUGE PROOF,
+  ... }
+```
+With this method only one line needs to be changed, syntax highlighting is usually preserved, and jumping to definition still works. It also has the advantage that it is extremely easy to remove all instances of `sorry;` from a file when finished using find and replace.
+
+### Memory and deterministic timeout
+Lean has two configurable flags for configuring its resource usage, memory and execution time.
+While the defaults for these are usually fairly sensible, when writing proofs it can be helpful to increase these options, if deterministic timeouts or out of memory issues slow down development.
+This is especially helpful when using tactics like `library_search` that even if they take a long time will be replaced by a faster tactic if they succeed.
+
+To change these settings we can add the flags `-M  <some number of megabytes>` (default 4096) for the memory limit and `-T <number of allocations>` (default 100000).
+In VSCode this can be configured by opening the settings and searching for `Lean Memory Limit` and `Lean Time Limit`.
+
+
+### Old mode
+If you change a file early on in the import hierarchy in a large project is changed and then attempt to work on a file that depends on it much later, then Lean will try to check all files in between to ensure they didn't break.
+This behaviour can cause the Lean server to become unresponsive, or simply slow down development when.
+You can also call lean with the "--old" flag to help in these situations, though it does come with some caveats.
```

Comment:
Was there some first suggestion that was to go before this? Otherwise I think:

```suggestion
You can call lean with the "--old" flag to help in these situations, though it does come with some caveats.
```

Reply:
Nope just a copy-paste error, thanks for the fixes!